### ISSUE TITLE
Possible deprecation method warnings React 18

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#9845]
We get the same warning as in https://github.com/handsontable/handsontable/issues/8748 but now using React 18.
They are shown once we render the Handsontable component.

**Warnings**
```js
react-dom.development.js:86 Warning: Using UNSAFE_componentWillMount in strict mode is not recommended and may indicate bugs in your code. See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move code with side effects to componentDidMount, and set initial state in the constructor.

Please update the following components: HotTable
```
```js
Warning: Using UNSAFE_componentWillUpdate in strict mode is not recommended and may indicate bugs in your code. See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.

Please update the following components: HotTable
```

Related comment:
* https://github.com/handsontable/handsontable/issues/8748#issuecomment-1086114639 

### Your environment
* Handsontable version: 12.1.2
* Browser name and version: Chrome 104
* Operating system: macOS Monterey

**Comment by AMBudnik** [comment]
I was just testing a new playground and created this example https://handsontable-basic-load--aleksandrabudni.repl.co/ when I spotted that we get a reference to `HotTable2` component in the warning, while the component itself is still `HotTable` (to check the code go to https://replit.com/@AleksandraBudni/Handsontable-basic-load#src/App.jsx

**Comment by ottosson** [comment]
I get this error message but with HotTable2:

![image](https://user-images.githubusercontent.com/3355320/200196338-10ad075f-3f45-4c25-a0c7-6a6c1220df11.png)

is this affecting anything important or can I just ignore it for the moment? It would be nice if it could be fixed still.

**Comment by AMBudnik** [comment]
Hi @ottosson 

This is only a warning. But here on the board, we marked that as a `change` that should be altered. 
I will ping you as soon as we make the necessary changes.

**Comment by AMBudnik** [comment]
Hi @ottosson 

I am happy to announce that this issue is fixed in the latest version - Handsontable v12.3.3
Here you can read more about the changes https://handsontable.com/docs/javascript-data-grid/release-notes/
